### PR TITLE
weapon hit code cleaning

### PR DESCRIPTION
**Pull request by stupidnewbie**
I tweaked the weighted random code slightly.  I believe weapons will hit at their intended rate regardless of their accuracy with this change.

I also changed the code to state things in terms of affirmation.  A hit is true and a miss is false.  Hopefully this is more direct and easier to read.

### CODE REVIEW

File: lib/Default/WeightedRandom.class.inc [hunk 1]
```diff
@@ -73,19 +73,15 @@ class WeightedRandom {
...
-		$failed = mt_rand(0, 99) + $this->weighting >= $percent;
-		$this->adjustWeightByPercent(!$failed, $percent);
-		return $failed;
-	}
-	
-	public function adjustWeightByPercent($success, $percent) {
-		$percent = $percent / 100;
-		if ($success) {
-			$percent = -1 + $percent;
+	public function checkHit($weaponAccuracy) {
```
Comment (Page-) [inline lib/Default/WeightedRandom.class.inc]: This code looks equivalent to before but just inverted to me? Am I missing something as you mentioned you tweaked it

Comment (stupidnewbie) [reply lib/Default/WeightedRandom.class.inc]: I believe you are correct.  Looking back I suspect my testing was at fault for the numbers I was getting.

File: lib/Default/WeightedRandom.class.inc [hunk 2]
```diff
@@ -73,19 +73,14 @@ class WeightedRandom {
...
```
Comment (hemberger) [inline lib/Default/WeightedRandom.class.inc]: This class was previously context-agnostic. Is it possible to rename the function and variables to maintain that? (Perhaps something like `checkSuccess`, `$percentChance`, and `$success` or whatever seems suitable.)

File: lib/Default/SmrWeapon.class.inc [hunk 3]
```diff
@@ -249,44 +245,52 @@ class SmrWeapon extends AbstractSmrCombatWeapon {
 		return $damage;
 	}
 	
+	public function checkHit(AbstractSmrPlayer $player, $weaponAccuracy) {
+		$this->getWeightedRandomForPlayer($player)->flipWeightedCoin($weaponAccuracy);
```
Comment (hemberger) [inline lib/Default/SmrWeapon.class.inc]: This is missing a return value. If you used this code in testing the changes, then you probably should have seen misses 100% of the time.

**Pull request merged**